samples/XAMLCatalog/XAMLCatalog/UIKitControls/UILabelViewController.m

```diff
@@ -14,351 +14,309 @@
-static const CGFloat c_originY = 8;
-static const CGFloat c_width = 300;
-static const CGFloat c_height = 50;
-static const CGFloat c_labelFontSize = 17.0f;
-static const int TAG_SUBVIEW_UILABEL = 1;
+#import <MobileCoreServices/MobileCoreServices.h>
 
 @implementation UILabelViewController {
-@private
-    NSMutableArray* _labels;
+    int _eventId;
 }
 
-- (UILabel*)_createUILabelWithColor:(UIColor*)color
-                               text:(NSString*)text
-                      textAlignment:(NSTextAlignment)alignment
-                      lineBreakMode:(UILineBreakMode)lineBreakMode
-                      numberOfLines:(NSInteger)numberOfLines {
-    CGRect frame = CGRectMake(c_originX, c_originY, c_width, c_height);
-    UILabel* label = [[UILabel alloc] initWithFrame:frame];
-    label.textColor = color;
-    label.text = text;
-    label.textAlignment = alignment;
-    label.lineBreakMode = lineBreakMode;
-    label.numberOfLines = numberOfLines;
-    return label;
+// setting up the controls to update the UILabel config
+- (void)_setup {
+    self.text = [[UITextField alloc] initWithFrame:CGRectMake(105.0f, 20.0f, 216.0f, 30.0f)];
+    [self.view addSubview:self.text];
+
+    self.fontSize = [[UISlider alloc] initWithFrame:CGRectMake(101.0f, 54.0f, 187.0f, 31.0f)];
+    self.fontSize.minimumValue = 4.01f;
+    self.fontSize.maximumValue = 80;
+    [self.view addSubview:self.fontSize];
+
+    self.x = [[UITextField alloc] initWithFrame:CGRectMake(74.0f, 92.0f, 45.0f, 30.0f)];
+    [self.view addSubview:self.x];
+
+    self.y = [[UITextField alloc] initWithFrame:CGRectMake(141.0f, 92.0f, 45.0f, 30.0f)];
+    [self.view addSubview:self.y];
+
+    self.w = [[UITextField alloc] initWithFrame:CGRectMake(211.0f, 92.0f, 45.0f, 30.0f)];
+    [self.view addSubview:self.w];
+
+    self.h = [[UITextField alloc] initWithFrame:CGRectMake(293.0f, 92.0f, 45.0f, 30.0f)];
+    [self.view addSubview:self.h];
+
+    self.numberOfLines = [[UITextField alloc] initWithFrame:CGRectMake(104.0f, 132.0f, 32.0f, 30.0f)];
+    [self.view addSubview:self.numberOfLines];
+
+    self.lineBreakMode = [[UITextField alloc] initWithFrame:CGRectMake(207.0f, 132.0f, 32.0f, 30.0f)];
+    [self.view addSubview:self.lineBreakMode];
+
+    self.autoShrink = [[UISwitch alloc] initWithFrame:CGRectMake(310.0f, 131.0f, 51.0f, 31.0f)];
+    [self.view addSubview:self.autoShrink];
+
+    self.minimumFontSize = [[UITextField alloc] initWithFrame:CGRectMake(80.0f, 167.0f, 56.0f, 30.0f)];
+    [self.view addSubview:self.minimumFontSize];
+
+    self.minimumScaleFactor = [[UITextField alloc] initWithFrame:CGRectMake(239.0f, 167.0f, 41.0f, 30.0f)];
+    [self.view addSubview:self.minimumScaleFactor];
+
+    self.label = [[UILabel alloc] initWithFrame:CGRectMake(4.0f, 254.0f, 353.0f, 107.0f)];
+    self.label.text = @"This is a test string used to test UILabel - and it should wrap and can be as long as three lines";
+    self.label.backgroundColor = [UIColor redColor];
```

Comment:
color are added for visual cue to know where the boundary is. removed.  

---
In reply to: [104794999](https://github.com/Microsoft/WinObjC/pull/2178#discussion_r104794999) [](ancestors = 104794999)